File: RandomTests/Clause.java
class Clause {
  Clause(final int len, final int[] hgs, final int base, final int neck, final int[] xs) {
    // head+goals pointing to cells in cs
    this.hgs = hgs; 
    // heap where this starts
    this.base = base; 
    this.len = len; 
    // length of heap slice
    this.neck = neck; 
    // first after the end of the head
    this.xs = xs; 
    // indexables in head


    System.out.println("The Length value:"+len);
    System.out.println("The Base value:"+base);
    System.out.println("The Index value XS:");
    for(int i=0; i<xs.length;i++)
    System.out.print("  "+xs[i]);
    System.out.println();
    System.out.println("Value of the neck:"+neck);
    System.out.println("Heap: ");
    for(int i=0; i<hgs.length;i++)
        System.out.print("  "+hgs[i]);
  }
  final int len;
  final int[] hgs;
  final int base;
  final int neck;
  final int[] xs;

  }

public class ClauseTest{
	public static void main(String args[])
	{
	    int ab[]; ab=new int[3];int bc[]; bc=new int[3];
    //defining seed
        int seed=35;
	   
	    ab=random(34);
	
		bc=random(35);
		Clause cl=new Clause(seed+10,bc,seed+12,seed+35,ab);
	}
	
public static int[] random(int seed){
       int[] data;int random;
        data=new int[3];
         
         for (int i=100,j=0;j<3;i=i*seed,j++){
            //random value is defined as follows
            random=i*seed+seed*(i-seed);
            data[j]=random;

        }
        return data;

}

}

File: clause.cpp
#include <algorithm>
#include <cstdlib>
#include <stack>
#include <cmath>
#include <iostream>
#include <bits/stdc++.h>
using namespace std;
/*
 ***************The clause is represnted in the following manner******************
 */
class Clause
{
public:
  int len;
  int * hgs;
  int base;
  int neck;
  int * xs;

  Clause( int len, int hgs[], int base, int neck,int xs[])
  {
   // head+goals pointing to cells in cs
    this->hgs=hgs;  
    // heap where this starts
    this->base=base; 
    // length of heap slice
    this->len=len; 
    // first after the end of the head
    this->neck=neck;
    // indexables in head
    this->xs= xs; 
    cout<<"The value of Length :";
    cout<<len;
    cout<<"\n";
    cout<<"The value of Base is: ";
    cout<<base;
    cout<<"\n";
    cout<<"The Index xs is : \n";
    for(int i=0;i<3;i++)
{
    cout << hgs[i];
    cout<<" ";

}
  cout<<"\n";
  cout<<"The Neck value is : ";
  cout<<neck;
   cout<<"\n";
    cout<<"The Heap value is : \n";
    for(int j=0;j<3;j++)
{
    cout << xs[j];
    cout<<" ";


}
}
};

int * randomfun(int seed){
        static int data[3];int random;

        for (int i=100,j=0;j<3;i=i*seed,j++){
            
            // defining random values 
            random=i*seed+seed*(i-seed);
            //assigning data value to random value 
            data[j]=random;
        }
        return data;
}

int main()

{
 
 int *str1;int *str2;
 int d1[4];
 int seed=35;
  //defining str1
  str1=randomfun(34);
  
  for (int h=0;h<3;h++)
  {
    
      d1[h]=str1[h];
  }
  
//defining str2  
str2=randomfun(35);
Clause b(seed+10,d1,seed+12,seed+35,str1);
return 0;

  
}